commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/TwoSidedDrive.scala

```diff
@@ -32,46 +55,35 @@ trait TwoSidedDrive extends UnicycleDrive { self =>
     )
   }
 
-  protected def expectedVelocity(drive: TwoSidedSignal): TwoSidedVelocity = {
-    TwoSidedVelocity(maxLeftVelocity * drive.left, maxRightVelocity * drive.right)
+  protected def expectedVelocity(drive: TwoSidedSignal)(implicit props: DrivetrainProperties): TwoSidedVelocity = {
+    TwoSidedVelocity(props.maxLeftVelocity * drive.left, props.maxRightVelocity * drive.right)
   }
 
-  protected def driveClosedLoop(signal: SignalLike[TwoSidedSignal]): PeriodicSignal[TwoSidedSignal] =
-    TwoSidedControllers.velocity(signal.map(expectedVelocity))
-
-  protected def driveVelocity(signal: SignalLike[TwoSidedVelocity]): PeriodicSignal[TwoSidedSignal] =
-    TwoSidedControllers.velocity(signal)
-
-  protected val maxLeftVelocity: Velocity
-  protected val maxRightVelocity: Velocity
-
-  protected val leftControlGains: PIDFProperUnitsConfig[Velocity, Acceleration, Length, Dimensionless]
-  protected val rightControlGains: PIDFProperUnitsConfig[Velocity, Acceleration, Length, Dimensionless]
-
-  // disable unicycle forward control because we already control that here
-  override protected val forwardControlGains: PIDFConfig[Velocity, Velocity, Acceleration, Length, Dimensionless] = PIDFConfig(
-    Each(0) / MetersPerSecond(1),
-    Each(0) / MetersPerSecondSquared(1),
-    Each(0) / Meters(1),
-    Each(0) / MetersPerSecond(0)
-  )
-
-  protected val leftVelocity: PeriodicSignal[Velocity]
-  protected val rightVelocity: PeriodicSignal[Velocity]
-
-  override protected val forwardVelocity: PeriodicSignal[Velocity] =
-    leftVelocity.zip(rightVelocity).map(t => (t._1 + t._2) / 2)
+  protected def driveClosedLoop(signal: SignalLike[TwoSidedSignal])
+                               (implicit hardware: DrivetrainHardware,
+                                         props: DrivetrainProperties): PeriodicSignal[TwoSidedSignal] =
+    TwoSidedControllers.closedLoopControl(signal)
 
   object TwoSidedControllers {
```

Comment:
See above, we want it as an object since the trait will be mixed into a package. So this object will be more like a proper object inside the package instead of an object inside of a class.